migrations/admin_events_test.go

```diff
@@ -0,0 +1,150 @@
+package migrations
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	evgdb "github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/evergreen/mock"
+	"github.com/evergreen-ci/evergreen/testutil"
+	"github.com/mongodb/anser"
+	"github.com/mongodb/anser/db"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"gopkg.in/mgo.v2/bson"
+)
+
+type adminEventSuite struct {
+	env      *mock.Environment
+	session  db.Session
+	database string
+	now      time.Time
+	cancel   func()
+	suite.Suite
+}
+
+func TestAdminEventMigration(t *testing.T) {
+	require := require.New(t)
+
+	mgoSession, database, err := evgdb.GetGlobalSessionFactory().GetSession()
+	require.NoError(err)
+	defer mgoSession.Close()
+
+	session := db.WrapSession(mgoSession.Copy())
+	defer session.Close()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	s := &adminEventSuite{
+		env:      &mock.Environment{},
+		session:  session,
+		database: database.Name,
+		cancel:   cancel,
+	}
+
+	require.NoError(s.env.Configure(ctx, filepath.Join(evergreen.FindEvergreenHome(), testutil.TestDir, testutil.TestSettings)))
+	require.NoError(s.env.LocalQueue().Start(ctx))
+
+	anser.ResetEnvironment()
+	require.NoError(anser.GetEnvironment().Setup(s.env.LocalQueue(), s.session))
+	anser.GetEnvironment().RegisterCloser(func() error { cancel(); return nil })
+
+	suite.Run(t, s)
+}
+
+func (s *adminEventSuite) SetupTest() {
+	s.NoError(evgdb.ClearCollections(eventCollection))
+	s.now = time.Now().Round(time.Millisecond).Truncate(time.Millisecond)
+
+	data := bson.M{
```

Comment:
Switch these bson.M's to anser db.Document.

Reply:
done